api/main_endpoints/routes/User.js

```diff
@@ -509,4 +509,41 @@ router.post('/usersSubscribedAndVerified', function(req, res) {
     });
 });
 
+// For Club Revenue page 
+router.post('/countMembers', async (req, res) => {
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(FORBIDDEN);
+  } else if (!checkIfTokenValid(req)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+  
+  const currentYear = new Date().getFullYear();
+  const today = new Date();
+
+  let beginningOfSemester, endOfThisSemester, endOfNextSemester;
+
+  if (today < new Date(currentYear, 5, 1)) {
+    beginningOfSemester = new Date(currentYear, 0, 1); 
+    endOfThisSemester = new Date(currentYear + 1, 0, 1); 
+    endOfNextSemester = new Date(currentYear + 1, 0, 1);
+  } else {
+    beginningOfSemester = new Date(currentYear, 5, 1); 
+    endOfThisSemester = new Date(currentYear + 1, 0, 1); 
+    endOfNextSemester = new Date(currentYear + 1, 5, 1);
+  }
```

Comment:
make the initial values of the variables these, put it above the if statement, no need for else